Oppdater til capacitor v6

jorgkv opened the pull request:
Påkrevd oppdatering som må tas inn for å kunne rulle ut på android fra og med august.

En av commitene deaktiverer / kommenterer ut offline-registreringer ved hjelp av sqlite, siden [sqlite-plugin ikke støtter capacitor 6 (enda)](https://github.com/capacitor-community/sqlite/issues/541).

Denne endringen påvirker native plugins, i alle fall capacitor-plugins. Cordova-plugins er ikke påvirket tror jeg. Vi bør derfor teste kamera, filsystem-kode (offline-kart), osv.

Har notert meg at vi bør teste kamera / legge til bilder på en android-enhet med API < 30 / mindre enn android 11.0 (i tillegg til nyere systemer selvsagt), fordi det er en annen type håndtering av bilder/kamera der.

Etter at vi fiksa å legge til bilder på Android 13, må vi teste dette på nytt iOS.

Jeg har testet at appen starter på android-emulatorer med API 34 og 30.

jorgkv committed:
Oppdater til capacitor 6

  - Upgrade to latest capacitor cli

  - Run npx cap migrate

  - Run npm install with updated dependencies

  - Oppdateringer etter npx cap sync

  - Run AGP Upgrade Assistant

  - Upgrade AGP dependency from 8.1.1 to 8.3.2

  - Upgrade Gradle version to 8.4

jorgkv committed:
Fjern NSLocationAlwaysUsageDescription

Denne beskrivelsen er deprecated og kan fjernes.

jorgkv committed:
Kommenter ut offline-registreringer

Dette er for å kunne teste appen inntil
sqlite-plugin kan brukes med capacitor v6.

NB: Hvis denne commiten revereses, husk å kjøre
npm install i tillegg til det som er gjort her for
å få med endringer i package-lock.json også.

gruble commented:
Denne blei ganske masete etter at jeg satte tlf i flymodus. Kommer opp ganske ofte:
![image](https://github.com/NVE/regObs4/assets/71138449/65f2a70d-3e3c-4b3e-b1d8-5f31aab3a7b8)

gruble commented:
Får til å ta bilder med kamera og ta dem rett inn i appen, men jeg får ikke til å legge til bilder fra album:
![image](https://github.com/NVE/regObs4/assets/71138449/97571d34-1231-4778-9512-fee4e16651cc)

gruble commented:
Fant også denne i konsollet, men tror ikke det har noe med endringer i denne PR'en å gjøre. Har laget en [egen sak](https://nveprojects.atlassian.net/browse/RO-2635) på det:
```
Error: NG0100: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checked. Previous value: 'null'. Current value: '0'
...
at OfflineMapPage_ion_footer_10_Template (offline-map.page.html:50:20)
```

gruble reviewed (changes requested):
Tror jeg har testet alle plugins på min Android 12-tlf nå. 
Det eneste som ikke funka er kamera-plugin'en, så det må vi vel fikse. Hvis ikke du får til å gjenskape det, kan jeg godt se på det. Og den feilmeldinga om observasjoner man ikke får lastet ned, kan vi kanskje kneble litt, særlig når vi er offline?

gruble commented:
Jeg lurer på om det er spesialtegnet %3A i filnavnet den har problemer med. Vi hadde jo en [sak](https://nveprojects.atlassian.net/browse/RO-1456) tidligere om at den ikke liker mellomrom i filnavnet? Jeg sjekker litt nærmere.
Har testet å legge til bilder fra album nå i 4.8.12 og der funker det fint.

Ser ut som det er [en del endringer i kamera-plugin'en](https://capacitorjs.com/docs/apis/camera#android) i Capacitor 6 for Android. Den funker f.eks. ikke på Android 10 og lavere, uten å legge til noen egne innstillinger i manifest-fila. Sjekket Plausible, og [vi har fortsatt en del brukere med Android 10 og lavere](https://plausible.io/regobs.no?props=%7B%22platform%22%3A%22app%22%7D&os=Android).

jorgkv commented:
> Ser ut som det er [en del endringer i kamera-plugin'en](https://capacitorjs.com/docs/apis/camera#android) i Capacitor 6 for Android. Den funker f.eks. ikke på Android 10 og lavere, uten å legge til noen egne innstillinger i manifest-fila. Sjekket Plausible, og [vi har fortsatt en del brukere med Android 10 og lavere](https://plausible.io/regobs.no?props=%7B%22platform%22%3A%22app%22%7D&os=Android).

Jepp, så det lar seg løse. Jeg tenkte bare det var lurt å teste på en emulator / enhet med android 10 eller lavere før vi faktisk gjorde det som står der.

gruble committed:
Revert "Kommenter ut offline-registreringer"

This reverts commit f26509403e59ec1654cc8b07f0ef3d3f51e842d6.

gruble committed:
Oppgradert @capacitor-community/sqlite til 6.0.0-alpha.1

Dette er første versjon som støtter Capacitor 6

gruble committed:
Fiks av lagring av observasjoner i offline-databasen med tekst som inneholder anførstelstegn

Se RO-2656

gruble committed:
Nå kan vi igjen legge til bilder fra album på Android.

Har tatt i bruk Capacitor Filesystem API i stedet for @awesome-cordova-plugins/file for å kopiere fila fra kamera-plugin

gruble committed:
Lagt til støtte for bruk av fotoalbum  for gamle Android-versjoner

gruble committed:
Bedre dokumentasjon på addAttchmentAsUrl()

gruble committed:
Lagt til SqlLite-plugin for iOS

av en eller annen grunn legger Capacitor inn dette to ganger

gruble committed:
JEkket opp versjonsnummer og kjørt enkel npm audit
